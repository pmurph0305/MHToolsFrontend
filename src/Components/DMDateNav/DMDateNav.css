.dateContainer {
    display:flex;
    justify-content: space-between;
    border-bottom: 0.1rem solid grey;
}