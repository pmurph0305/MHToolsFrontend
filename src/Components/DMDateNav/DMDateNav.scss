@import '../../styles/shared.scss';

.dateContainer {
    display:flex;
    justify-content: space-between;
    border-bottom: 0.1rem solid grey;
}

.DateNavButton {
    @extend %ButtonBase;
    padding: 0.5rem;
    margin-bottom: 0.5rem;
    width: 6rem;
}

.DateDisplay {
    font-size: 1.5rem;
    padding: 0.5rem;
}