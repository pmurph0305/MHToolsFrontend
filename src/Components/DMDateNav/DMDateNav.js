import React from 'react';
import './DMDateNav.css'

const DMDateNav = ({date, onClick}) => {
    return (
        <div className="dateContainer">
            <button
                value="-1"
                onClick={onClick}
                className="f6 dim ph3 pv2 mb2 dib white bg-black pointer"
                
            >Previous</button>
            <p className="f3">{date}</p>
            <button 
                value="1"
                onClick={onClick}
                className="f6 dim ph3 pv2 mb2 dib white bg-black pointer"
            >Next</button>
        </div>
    )
}

export default DMDateNav;