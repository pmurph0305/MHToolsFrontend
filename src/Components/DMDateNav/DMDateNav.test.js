import { shallow } from 'enzyme';
import React from 'react';
import DMDateNav from './DMDateNav'


describe('DMDateNav tests', () => {
    let wrapper;
    let mockDate = '01-01-0101'
    beforeEach(() => {
        wrapper = shallow(<DMDateNav date={mockDate}/>)
    })

    it('Exists', () => {
        expect(wrapper.exists()).toBe(true);
    })

    it('Matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('Displays date correctly', () => {
        expect(wrapper.find('p').text()).toBe(mockDate)
    })

    it('On Click works', () => {
        let mockClick = jest.fn();
        let wrapClick = shallow(<DMDateNav onClick={mockClick}/>)
        wrapClick.find('button').first().simulate('click');
        expect(mockClick).toHaveBeenCalledTimes(1);
        wrapClick.find('button').last().simulate('click');
        expect(mockClick).toHaveBeenCalledTimes(2);
    })
})