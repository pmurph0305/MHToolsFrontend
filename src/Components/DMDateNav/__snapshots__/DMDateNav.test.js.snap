// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DMDateNav tests Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DMDateNav
    date="01-01-0101"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="f6 dim ph3 pv2 mb2 dib white bg-black pointer"
          value="-1"
        >
          Previous
        </button>,
        <p
          className="f3"
        >
          01-01-0101
        </p>,
        <button
          className="f6 dim ph3 pv2 mb2 dib white bg-black pointer"
          value="1"
        >
          Next
        </button>,
      ],
      "className": "dateContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Previous",
          "className": "f6 dim ph3 pv2 mb2 dib white bg-black pointer",
          "onClick": undefined,
          "value": "-1",
        },
        "ref": null,
        "rendered": "Previous",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "01-01-0101",
          "className": "f3",
        },
        "ref": null,
        "rendered": "01-01-0101",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "className": "f6 dim ph3 pv2 mb2 dib white bg-black pointer",
          "onClick": undefined,
          "value": "1",
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="f6 dim ph3 pv2 mb2 dib white bg-black pointer"
            value="-1"
          >
            Previous
          </button>,
          <p
            className="f3"
          >
            01-01-0101
          </p>,
          <button
            className="f6 dim ph3 pv2 mb2 dib white bg-black pointer"
            value="1"
          >
            Next
          </button>,
        ],
        "className": "dateContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Previous",
            "className": "f6 dim ph3 pv2 mb2 dib white bg-black pointer",
            "onClick": undefined,
            "value": "-1",
          },
          "ref": null,
          "rendered": "Previous",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "01-01-0101",
            "className": "f3",
          },
          "ref": null,
          "rendered": "01-01-0101",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "f6 dim ph3 pv2 mb2 dib white bg-black pointer",
            "onClick": undefined,
            "value": "1",
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
